src/axom/primal/operators/clip.hpp

```diff
@@ -82,14 +79,13 @@ Polygon<T, 3> clip(const Triangle<T, 3>& tri, const BoundingBox<T, 3>& bbox)
   {
     // Optimization note: we should be able to save some work based on
     // the clipping plane and the triangle's bounding box
-
-    if(triBox.getMax()[dim] > bbox.getMin()[dim])
+    if(triBox.getMin()[dim] < bbox.getMin()[dim])
```

Comment:
Why the refactor, since both old and new versions are correct?  Is this clearer to you?  (Also (new) line 88)

Reply:
This change and the one a few lines down were the actual bugfix  -- see #1239 
The rest of this PR was adding/improving the tests and some improvements along the way.

I'm actively investigating a change to the In/Out regression tests where some points near the surface boundary have changed their in/out status based on this change to `clip(Triangle, BoundingBox)` -- 
E.g. there are 4 changed points in the cap.stl on a 32^3 grid
```
2024-01-02T21:43:48.9304742Z 313: Test command: /usr/bin/mpirun "-np" "1" "/home/axom/axom/build-clang@10.0.0-release/tests/quest_regression_test" "--mesh" "/home/axom/axom/data/quest/cap.stl" "--baseline" "/home/axom/axom/data/quest/regression/cap_32_baseline.root"
...
2024-01-02T21:43:50.3919836Z 313: [INFO] ** Containment test failed.  There were 4 differences. Showing first 4
2024-01-02T21:43:50.3922470Z 313:   Disagreement on sample 5725 @ (0,-2,-2.75).  Expected 0, got 1
2024-01-02T21:43:50.3925037Z 313:   Disagreement on sample 6946 @ (0,-1,-2.5).  Expected 1, got 0
2024-01-02T21:43:50.3927155Z 313:   Disagreement on sample 13942 @ (0,2.5,-1).  Expected 1, got 0
2024-01-02T21:43:50.3929419Z 313:   Disagreement on sample 26317 @ (0,-2.75,2).  Expected 0, got 1 
2024-01-02T21:43:50.3934394Z 313: [INFO] ** Baseline tests failed 
```
See:  https://dev.azure.com/healy22/d4445b8f-6e95-49cd-9e5b-8c8ca1ccad31/_apis/build/builds/9194/logs/77

These appear to be due to the check here: 
here: https://github.com/LLNL/axom/blob/develop/src/axom/quest/InOutOctree.hpp#L1138-L1156

It's not yet clear to me if the changes are acceptable or if I'll need to add some additional parameters to `clip()` to control whether/how clipping should handle fuzzy boundaries or if we should just update the regression tests.
(That's, e.g. what led to my refactoring of the fuzzy comparators into a separate file).

Reply:
**Update:** After digging into it further, I don't think we need to modify `clip` to handle fuzzy boundaries. 

There are very few changes in the quest_regression suite, which compares the containment determination between the octree-based `InOutOctree` and the bvh-based `SignedDistance` queries. All the changes are for points that lie _on_ the surface, where the containment determination changed due to the changes to `clip(Triangle, BoundingBox)`

When I run the full regression suite, I get
```
>ctest -R regression 
Test project <axom>/build-axom
      Start 318: quest_regression_sphere_16
 1/19 Test #318: quest_regression_sphere_16 ...............   Passed  129.96 sec
      Start 319: quest_regression_tetrahedron_31
 2/19 Test #319: quest_regression_tetrahedron_31 ..........   Passed    0.77 sec
      Start 320: quest_regression_aatbase_3_binary_16
 3/19 Test #320: quest_regression_aatbase_3_binary_16 .....   Passed    1.60 sec
      Start 321: quest_regression_aatbase_3_binary_32
 4/19 Test #321: quest_regression_aatbase_3_binary_32 .....   Passed    2.01 sec
      Start 322: quest_regression_boxedSphere_40
 5/19 Test #322: quest_regression_boxedSphere_40 ..........   Passed    5.28 sec
      Start 323: quest_regression_cap_16
 6/19 Test #323: quest_regression_cap_16 ..................***Failed    0.92 sec
      Start 324: quest_regression_cap_32
 7/19 Test #324: quest_regression_cap_32 ..................***Failed    1.16 sec
      Start 325: quest_regression_fin0_32
 8/19 Test #325: quest_regression_fin0_32 .................   Passed    0.89 sec
      Start 326: quest_regression_fin1_32
 9/19 Test #326: quest_regression_fin1_32 .................   Passed    0.86 sec
      Start 327: quest_regression_fin2_32
10/19 Test #327: quest_regression_fin2_32 .................   Passed    0.87 sec
      Start 328: quest_regression_naca0012_16
11/19 Test #328: quest_regression_naca0012_16 .............   Passed    1.00 sec
      Start 329: quest_regression_naca0012_32
12/19 Test #329: quest_regression_naca0012_32 .............   Passed    1.25 sec
      Start 330: quest_regression_plane_binary_16
13/19 Test #330: quest_regression_plane_binary_16 .........   Passed   15.35 sec
      Start 331: quest_regression_plane_binary_32
14/19 Test #331: quest_regression_plane_binary_32 .........   Passed   17.76 sec
      Start 332: quest_regression_plane_simp_16
15/19 Test #332: quest_regression_plane_simp_16 ...........   Passed    1.90 sec
      Start 333: quest_regression_plane_simp_32
16/19 Test #333: quest_regression_plane_simp_32 ...........   Passed    2.83 sec
      Start 334: quest_regression_plane_simp_120_120_60
17/19 Test #334: quest_regression_plane_simp_120_120_60 ...   Passed   24.20 sec
      Start 335: quest_regression_sphere_32
18/19 Test #335: quest_regression_sphere_32 ...............   Passed    1.22 sec
      Start 336: quest_regression_tetrahedron_100
19/19 Test #336: quest_regression_tetrahedron_100 .........***Failed    1.45 sec

84% tests passed, 3 tests failed out of 19

Total Test time (real) = 211.35 sec

The following tests FAILED:
	323 - quest_regression_cap_16 (Failed)
	324 - quest_regression_cap_32 (Failed)
	336 - quest_regression_tetrahedron_100 (Failed)
```

Digging further (with some additional debugging info):
```
>ctest -R quest_regression_cap_16 -VV
...
323: [INFO] ** Containment test failed.  There were 2 differences. Showing first 2 -- 
323:   Disagreement on sample 977 @ (0,-1,-2.5).  Expected inside, got outside -- distance from query point to surface is 0 
323:   Disagreement on sample 1963 @ (0,2.5,-1).  Expected inside, got outside -- distance from query point to surface is 0  
323: [INFO] ** Baseline tests failed 
```
and similarly:
```
>ctest -R quest_regression_cap_32 -VV
...
324: [INFO] ** Containment test failed.  There were 4 differences. Showing first 4 -- 
324:   Disagreement on sample 5725 @ (0,-2,-2.75).  Expected outside, got inside -- distance from query point to surface is 0 
324:   Disagreement on sample 6946 @ (0,-1,-2.5).  Expected inside, got outside -- distance from query point to surface is 0 
324:   Disagreement on sample 13942 @ (0,2.5,-1).  Expected inside, got outside -- distance from query point to surface is 0 
324:   Disagreement on sample 26317 @ (0,-2.75,2).  Expected outside, got inside -- distance from query point to surface is 0  
```
and
```
>ctest -R quest_regression_tetrahedron_100 -VV
...
336: [INFO] ** Containment test failed.  There were 6 differences. Showing first 6 -- 
336:   Disagreement on sample 514344 @ (0.2,-0.8,0).  Expected inside, got outside -- distance from query point to surface is 0 
336:   Disagreement on sample 514752 @ (0.6,-0.4,0).  Expected outside, got inside -- distance from query point to surface is 0 
336:   Disagreement on sample 534948 @ (0.2,-0.6,0.2).  Expected outside, got inside -- distance from query point to surface is 0 
336:   Disagreement on sample 535748 @ (-0.6,0.2,0.2).  Expected outside, got inside -- distance from query point to surface is 0 
336:   Disagreement on sample 555948 @ (-0.6,0,0.4).  Expected outside, got inside -- distance from query point to surface is 0 
336:   Disagreement on sample 576352 @ (-0.4,0,0.6).  Expected inside, got outside -- distance from query point to surface is 0  
```

I'm going to update the regression baselines in the `axom_data` repo and merge this.

Reply:
Note: Issue https://github.com/LLNL/axom/issues/611 relates to consistency in the InOut determination for points on the boundary.